### ISSUE TITLE
Add option to C-BOT to evaluate 1 as true and 0 as false in while.

### ISSUE THREAD

**Issue by realszopen** [colobot/colobot#1272]
It would be nice if there was option to write while(1) or while(0) instead of while(true) or while(false).

**Comment by V-2Schneider** [comment]
Furthermore, all non-zero values could be treated as true

**Comment by MrSimbax** [comment]
I don't agree with this proposal. Such a feature encourages bad programming practices and most languages other than C/C++ do not allow such a conversion if I recall correctly.

**Comment by Erihel** [comment]
Not only C/C++ does it. Python does even more. It evaluates empty string/list/dict as false and true if they hold anything.

**Comment by MrSimbax** [comment]
Ok, but I still think such a feature does more harm than good. I would like to see examples where it is needed or helps readability. `while(1)` is a bad example IMO.

**Comment by Erihel** [comment]
`while(1)` is a bad example. Still it is readable. How about reading a file in parts?

```c
while (fread(buffer, 1, 512, fp)) {
	...
}
```
Here's something a bit different. It's operator overloading but still it's not explicit check https://github.com/colobot/colobot/blob/master/test/cbot/compile_graph.cpp#L49

**Comment by MrSimbax** [comment]
To be honest, such loops have always been confusing for me. I'd rather have the condition stated explicitly unless the function clearly returns boolean (which usually means its name indicates it), otherwise the whole statement looks suspicious and almost always makes me stop to ponder why they work. Maybe they're easier to write but certainly not easier to read, at least not for me. They just hide too many details and feel too much like magic. Maybe I'm alone with these feelings.

And `while(1)` may be readable because it's simple enough, but I still prefer `while(true)` since it feels less like a hack and reads better in plain English, hence communicates the intent more clearly. I've rarely seen such constructs in pseudocode which I think says quite a lot about their (un)readability.

**Comment by Bartolomeus-649** [comment]
It should be a bool.

If anything but 0 and false is interrelated as true, then it is easy to make mistakes like `if(a=4) ... `, which will produce a compilation error when a bool is required.

Also, the language should help/promote writing code that minimize bugs, avoid ambiguities and increase understanding the intention of the code.

And lets face it, writing the extra comparison logic is not really a thing that affect coding productivity that much....never heard someone say: "the whole project is late just because we had to use bool-values for iwhile-statements".

**Comment by melex750** [comment]
```
for(;;) // this works instead of while(true)
{
}
```

**Comment by hexagonrecursion** [comment]
> ```
> for(;;) // this works instead of while(true)
> {
> }
> ```

Irrelevant. This issue is about whether int should implicitly convert to bool.

The proposal is to allow writing `while(<expr>) { ... }` in place of `while(<expr> != 0) { ... }`.

**Comment by melex750** [comment]
I can understand your confusion if you didn't read the OP and take it literally.
